doc/devel/MEP/MEP29.rst

```diff
@@ -0,0 +1,133 @@
+========================
+Overhauling the Axes API
+========================
+
+.. contents::
+   :local:
+
+Abstract
+========
+The axes API at the moment feels bursting at the seems with so many tweaks and
+new features over the years that it has come time for a simplification.  We
+have many different types of Axis now, from Polar coordinates, Axes3D, and even
+Basemap.
+
+This MEP will simplify the API making our existing classes simpler and pave the
+way for more interesting exotic Axes, such as those found in non-Euclidean
+geometry, we already implement one such gemoetry, i.e. basemap which implements
+the simplest elliptic geometry, and provides various projections for which to
+project this geometry onto a 2D surface (the screen/paper).
+
+
+Detailed description
+====================
+Before we can look at changing the system, we first need to understand the
+concepts of Axes, and this we do in this section.
+
+What do we mean by Axes?
+------------------------
+Before we can look into the concept of Axes, we need to first look at the
+singular, at its simplest level, we think of an Axis as one dimensional,
+a simple number line.  It has units and it has a scale.
+
+It contains a coordinate space made of 1 or more `Axis` and we can plot in this
+defined space.
+
+Some examples of axes:
++ 2D Cartesian (x, y)
++ 2D Polar (rho, theta)
++ 3D Cartesian (x, y, z)
++ 3D Spherical Polar
++ 3D Cylindrical Polar
++ 2D on a sphere as we have in Basemap
+
+In fact Lagrangian mechanics simplifies the multitude of axes to a generic set
+of axis \vec{q}.  As we depend on 2D Cartesian geometry for output to the
+screen, and or paper documents etcetera, we thus need to convert from our axes to
+2D Cartesian coordinates.
+
+So as not to rewrite all the plot methods for every axes, we need our Base Axes
+class to do the conversion for us through an Axes API.
+
+Secondly, we need this API to convert from screen coordinates back to our Axes
+coordinates so as to facilitate user interaction.
+
+Finally, we need to stay aware of the fact that coordinate systems do not have
+a 1:1 mapping with screen coordinates, that we will specify extra parameters
+determined at run time to control this, for example at present the standard 2d
+axes uses four parameters to control the extent of the 2D Cartesian screen
+domain, set via x_lim, and y_lim these get affected by the pan/zoom controls
+of the GUI.  In 3d we have more parameters to control the rotation and zoom.
+
+As well as Axes 3D, Basemap should also welcome this change, with an
+anticipated structure of a base mapping class with a coordinate system in
+lat/lon coordinates, but with different mapping projections available for the
+conversion between the Axes coordinate system and the screen.
+
+Implementation
+==============
+Axes
+----
+
+First we define our coordinate transformation functions:
+axes_to_base(self, \*q)
+base_to_axes(self, x, y)
```

Comment:
These live in the transform in existing matplotlib speak. It might be worth mentioning this (or even providing a section on what the transform currently does).